Source/WebDriver/WebDriverService.cpp

```diff
@@ -475,46 +475,42 @@ void WebDriverService::handleMessage(WebSocketMessageHandler::Message&& message,
         return;
     }
 
+    const auto& asObject = parsedMessageValue->asObject();
+    if (!asObject) {
+        RELEASE_LOG_ERROR(WebDriverBiDi, "WebDriver handle BiDi message: Expected object.");
+        completionHandler(WebSocketMessageHandler::Message::fail(CommandResult::ErrorCode::InvalidArgument, connection));
+        return;
+    }
+
+    std::optional<int> commandId = asObject->getInteger("id"_s);
+    if (!commandId) {
+        RELEASE_LOG_ERROR(WebDriverBiDi, "Missing command ID.");
+        completionHandler(WebSocketMessageHandler::Message::fail(CommandResult::ErrorCode::InvalidArgument, connection, "Missing command id"_s));
+        return;
+    }
+
     BidiCommandHandler handler;
-    unsigned id = 0;
     RefPtr<JSON::Object> parameters;
-    if (!findBidiCommand(parsedMessageValue, &handler, id, parameters)) {
-        RELEASE_LOG(WebDriverBiDi, "Failed to find appropriate BiDi command");
-        std::optional<int> commandId;
-        if (auto parsedMessageObject = parsedMessageValue->asObject()) {
-            auto parsedCommandId = parsedMessageObject->getInteger("id"_s);
-            if (parsedCommandId && *parsedCommandId >= 0)
-                commandId = parsedCommandId;
-        }
-
-        auto errorCode = CommandResult::ErrorCode::UnknownCommand;
-        auto errorReply = WebSocketMessageHandler::Message::fail(errorCode, connection, { "Command not supported"_s }, commandId);
-        completionHandler(WTFMove(errorReply));
+    if (!findBidiCommand(asObject, &handler, parameters)) {
```

Comment:
Message parsing and the handler dispatcher can be generated from `BidiSession.json` similar to other Bidi modules, you shouldn't need to keep a static list of methods. Maybe add a FIXME for now?

Reply:
Yes, in the long run we could add similar dispatchers here too. It's a leftover from the original bidi prototype, based on the classic implementation. I'll link the static methods issue in the comment (https://bugs.webkit.org/show_bug.cgi?id=281721)